static/app/utils/withProjectsSpecified.tsx

```diff
@@ -34,6 +34,14 @@ const withProjectsSpecified = <P extends InjectedProjectsProps>(
       return ProjectsStore.getState(this.props.specificProjectSlugs);
     },
 
+    UNSAFE_componentWillReceiveProps(nextProps: Props) {
```

Comment:
@billyvg Would we prefer componentDidUpdate with the eslint-disable-line when setting the state inside it?

Reply:
I think really what we'd want is `getDerivedState` but this uses `createReactClass`, I'm going to guess that `DidUpdate` will probably cause some unexpected issues with state because of where it is in the lifecycle.

Reply:
We could make it not use createReactClass first :smile:

Reply:
Yeah, I agree. I can refactor the legacy `createReactClass` component in a follow-up.